tests/unit/test_library_ingress.py

```diff
@@ -1,50 +1,216 @@
+        )
+
+    def test_with_service_namespace(self):
+        """
+        arrange: given a configuration with the service namespace provided
+        act: when the charm is constructed with the configuration
+        assert: then the service namespace from the configuration is used.
+        """
+        service_namespace = "service namespace 1"
+        config_dict = {"service-namespace": service_namespace}
+
+        class CharmWithConfigDict(NginxIngressConsumerCharm):
+            def __init__(self, *args):
+                super().__init__(*args, config_dict=config_dict)
+
+        self.harness = Harness(CharmWithConfigDict, meta=META)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+        self.assertEqual(
+            self.harness.charm.ingress.config_dict["service-namespace"], service_namespace
+        )
+        self.assertEqual(self.harness.charm.ingress.config_dict["model"], service_namespace)
+
+    def test_with_previous_config(self):
+        """
+        arrange: given a configuration with all the items pre-standard
+        act: when the charm is constructed with the configuration
+        assert: then the configuration is stored with the previous and new keys.
+        """
+        service_hostname = "service hostname 1"
+        service_name = "service name 1"
+        service_port = "service port 1"
+        service_namespace = "service namespace 1"
+        config_dict = {
+            "service-hostname": service_hostname,
+            "service-name": service_name,
+            "service-port": service_port,
+            "service-namespace": service_namespace,
+        }
+
+        class CharmWithConfigDict(NginxIngressConsumerCharm):
+            def __init__(self, *args):
+                super().__init__(*args, config_dict=config_dict)
+
+        self.harness = Harness(CharmWithConfigDict, meta=META)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+        self.assertEqual(
+            self.harness.charm.ingress.config_dict["service-hostname"], service_hostname
+        )
+        self.assertEqual(self.harness.charm.ingress.config_dict["service-name"], service_name)
+        self.assertEqual(self.harness.charm.ingress.config_dict["service-port"], service_port)
+        self.assertEqual(
+            self.harness.charm.ingress.config_dict["service-namespace"], service_namespace
+        )
+
+        self.assertEqual(self.harness.charm.ingress.config_dict["host"], service_hostname)
+        self.assertEqual(self.harness.charm.ingress.config_dict["name"], service_name)
+        self.assertEqual(self.harness.charm.ingress.config_dict["port"], service_port)
+        self.assertEqual(self.harness.charm.ingress.config_dict["model"], service_namespace)
+
+    def test_with_new_config(self):
+        """
+        arrange: given a configuration with all the items pre-standard
+        act: when the charm is constructed with the configuration
```

Comment:
Should these two lines be different from the ones in `test_with_previous_config`?

Reply:
Thank you, fixed in next commit